src/coreclr/vm/instancecalli.cpp

```diff
@@ -0,0 +1,84 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "common.h"
+
+void MethodDesc::GenerateFunctionPointerCall(DynamicResolver** resolver, COR_ILMETHOD_DECODER** methodILDecoder)
+{
+    STANDARD_VM_CONTRACT;
+    _ASSERTE(resolver != NULL);
+    _ASSERTE(methodILDecoder != NULL);
+    _ASSERTE(*resolver == NULL && *methodILDecoder != NULL);
+    _ASSERTE(IsIL());
+    _ASSERTE(GetRVA() != 0);
+
+    // Intrinsic must be a static method.
+    _ASSERTE(IsStatic());
+
+    MetaSig declarationSig(this);
+    UINT argCount = declarationSig.NumFixedArgs();
+
+    // Intrinsic must have at least the "this" argument and the IntPtr function pointer argument.
+    _ASSERTE(argCount >= 2);
+
+    SigTypeContext genericContext;
+    ILStubLinker sl(
+        this->GetModule(),
+        GetSignature(),
+        &genericContext,
+        this,
+        (ILStubLinkerFlags)ILSTUB_LINKER_FLAG_NONE);
+
+    ILCodeStream* pCode = sl.NewCodeStream(ILStubLinker::kDispatch);
+
+    // Copy the existing signature to add HASTHIS and EXPLICITTHIS.
+    CorElementType fnType = declarationSig.NextArg();
+    _ASSERTE(fnType == ELEMENT_TYPE_FNPTR);
+    SigPointer spToken = declarationSig.GetArgProps();
+    spToken.GetByte(nullptr); // Skip the element type (ELEMENT_TYPE_FNPTR)
+    declarationSig.SkipArg();
+    SigPointer sigPtrTokenEnd = declarationSig.GetArgProps();
+
+    SigBuilder sigBuilder;
+    spToken.CopySignature(GetModule(), &sigBuilder, &sigPtrTokenEnd, IMAGE_CEE_CS_CALLCONV_HASTHIS | IMAGE_CEE_CS_CALLCONV_EXPLICITTHIS);
+
+    // Create a token for the new signature.
+    uint32_t sigLen;
+    PCCOR_SIGNATURE pSig = (PCCOR_SIGNATURE)sigBuilder.GetSignature((DWORD*)&sigLen);
+    mdToken fcnPtrToken = pCode->GetSigToken(pSig, sigLen);
+
+    // Load "this" and each argument to the function pointer.
+    for (UINT i = 1; i < argCount; ++i)
+        pCode->EmitLDARG(i);
+
+    // Load the function pointer and call it.
+    pCode->EmitLDARG(0);
+    pCode->EmitCALLI(fcnPtrToken, argCount - 2, declarationSig.IsReturnTypeVoid() ? 0 : 1);
```

Comment:
```suggestion
    pCode->EmitCALLI(fcnPtrToken, argCount - 1, declarationSig.IsReturnTypeVoid() ? 0 : 1);
```
I think this should be argCount - 1, ie "this" and each argument to the function pointer.